backend/src/LeanCode.AppRating/Processes/SendEmailOnLowRateSubmitted.cs

```diff
@@ -0,0 +1,46 @@
+using LeanCode.AppRating.Configuration;
+using LeanCode.AppRating.EmailViewModels;
+using LeanCode.SendGrid;
+using MassTransit;
+using SendGrid.Helpers.Mail;
+using Serilog;
+
+namespace LeanCode.AppRating.Processes;
+
+public class SendEmailOnLowRateSubmitted<TUserId> : IConsumer<RateSubmitted<TUserId>>
+    where TUserId : notnull, IEquatable<TUserId>
+{
+    private readonly ILogger logger = Log.ForContext<SendEmailOnLowRateSubmitted<TUserId>>();
+    private readonly SendGridRazorClient sendGridRazorClient;
+    private readonly AdminConfiguration adminConfiguration;
+
+    public SendEmailOnLowRateSubmitted(SendGridRazorClient sendGridRazorClient, AdminConfiguration adminConfiguration)
+    {
+        this.sendGridRazorClient = sendGridRazorClient;
+        this.adminConfiguration = adminConfiguration;
+    }
+
+    public async Task Consume(ConsumeContext<RateSubmitted<TUserId>> context)
+    {
+        if (context.Message.Rating > adminConfiguration.MaximalLowRatingInclusive)
+        {
+            logger.Debug("Rating is not considered low, skipping");
+            return;
+        }
+
+        var vm = new LowRateSubmittedEmail { UserId = context.Message.UserId.ToString() };
```

Comment:
Like I said earlier - this is quite low on the data. The e-mail would then be just link to the overall panel (you can't even point to a single rating, because you don't have full key here. This will be just slow to respond. The e-mail has to have some actionable information insinde.

Reply:
on the user story level we din't have any data about the rating, but sure I can extend this with more info

Reply:
Then I would say that the user story was _too basic_ (like, really, if I would get an e-mail that says nothing, I would ignore it).